src/parsimmon.js

```diff
@@ -612,6 +767,11 @@ function string(str) {
 
 function byte(b) {
   assertNumber(b);
+  if (b > 0xff) {
+    throw new Error(
+      "Value specified to byte constructor is larger in value than a single byte."
```

Comment:
it would be nice to print back out the value of `b` here and format it as a hex number, I think